Data/HashMap/Internal.hs

```diff
@@ -181,6 +181,10 @@ import qualified Data.List                   as List
 import qualified GHC.Exts                    as Exts
 import qualified Language.Haskell.TH.Syntax  as TH
 
+#if defined(__MHS__)
+import Data.Traversable
+#endif
```

Comment:
This is okay, but also blecherous. We actually want to switch on the "base library" variant, not the compiler. Do you know if there's been any progress on how to do that?

Reply:
I think the condition is totally acceptable for this PR; I just think it would be good to open up a community discussion about the broader issue.